### DISCUSSION TITLE
File not found error when file exists

### DISCUSSION THREAD

**Discussion by NetworkTest** [ipxe/ipxe#687] [General]
ipxe 1.0.0+git-20190125.36a4c85-5.1, dnsmasq  2.85-1, nginx 1.18.0-6.1+deb11u2, Debian 11

I'm getting a file-not-found error when executing an iPXE to install Proxmox VE 7.1 via PXE Boot.

The client complains like this:

http://172.31.5.25/proxmox/7.1/initrd... ok
"[http://172.31.5.25/proxmox/7.1/linux26 ](http://172.31.5.25/proxmox/7.1/linux26%20)"vga=791%20video=vesafb:ywrap,mtrr%20ramdisk_size=16777216%20rw%20quiet%20initrd=initrd"splash=silent"... No such file or directory (https://ipxe.org/2d12603b)
Could not boot image: No such file or directory (https://ipxe.org/2d12603b)

The server is a Debian 11 box running ipxe, dnsmasq, and nginx. The initrd and linux26 files live in /var/www/proxmox/7.1, are world readable. Both files are accessible using a web browser (including linux26, without options).

Thanks in advance for troubleshooting clues.

/etc/dnsmasq.conf:

log-queries
log-dhcp

port=0

listen-address=172.31.5.25
interface=ens32

enable-tftp
tftp-root=/tftpboot

dhcp-range=172.31.5.128,172.31.5.254,255.255.255.0,24h

dhcp-option=3,172.31.5.1

dhcp-option=15,ipmi.example.tld

dhcp-option=28,172.31.5.255

dhcp-option=26,9198

dhcp-match=set:ipxe,175

dhcp-vendorclass=BIOS,PXEClient:Arch:00000

dhcp-boot=tag:!ipxe,tag:BIOS,undionly.kpxe,172.31.5.25

dhcp-boot=tag:!ipxe,tag:!BIOS,ipxe.efi,172.31.5.25

dhcp-boot=tag:ipxe,/pxelinux.0

/tftpboot/pxelinux.0:

#!ipxe
dhcp
set webserver http://172.31.5.25/
set serverip http://172.31.5.25/
set pveversion 7.1
set opts "vga=791 video=vesafb:ywrap,mtrr ramdisk_size=16777216 rw quiet initrd=initrd"
menu Please choose an operating system to boot
item normal Install Proxmox
item debug Install Proxmox (Debug Mode)
choose --default normal --timeout 5000 target && goto ${target}
:debug
set kernel "${webserver}/proxmox/${pveversion}/linux26 ${opts} splash=verbose proxdebug"
goto init
:normal
set kernel "${webserver}/proxmox/${pveversion}/linux26 ${opts} splash=silent"
goto init
:init
initrd ${webserver}/proxmox/${pveversion}/initrd
chain ${kernel}

contents of /var/www/html/proxmox/7.1:

root@pxeboot:/tftpboot# ls -l /var/www/html/proxmox/7.1
total 1158896
-rw-r--r-- 1 root root 1176654336 Jul 20 19:26 initrd
-rwxr-xr-x 1 root root 10047424 Jul 20 19:26 linux26

**Comment by mcb30** [comment]
Remove the quotation marks and it should work

**Reply by NetworkTest** [reply]
Thanks for your reply. Sorry, but even without the quotes linux26 fails with the same error. In pxelinux.0 I tried removing the quotation marks both in the opts assignment and in the :normal section. 

What finally did work was formatting the opts variable in the form of PHP options, starting with a question mark and using ampersands instead of spaces as delimiters:

set opts ?vga=791&video=vesafb:ywrap,mtrr&ramdisk_size=16777216&rw&quiet&initrd=initrd&splash=silent

and then losing the quotes and spaces in the kernel line:

set kernel ${webserver}/proxmox/${pveversion}/linux26${opts}

Thanks again.

**Reply by NiKiZe** [reply]
You should not have VGA etc in the URL at all, you want that as a cmdline to the kernel, not as arguments to the server when requesting the file.
